partiql-spi/src/test/kotlin/org/partiql/spi/value/DatumToStringTest.kt

```diff
@@ -0,0 +1,280 @@
+package org.partiql.spi.value
+
+import au.com.origin.snapshots.Expect
+import au.com.origin.snapshots.junit5.SnapshotExtension
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.partiql.spi.types.PType
+import java.math.BigDecimal
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+import java.time.OffsetDateTime
+import java.time.OffsetTime
+import java.time.ZoneOffset
+
+/**
+ * Test cases for toString methods of all Datum implementations.
+ * Verifies that each Datum class produces expected string representations.
+ *
+ * Test methods are in alphabetical order by method name
```

Comment:
nit: let's add some brief docs on how to run the snapshot tests and update the results if there's a divergence